### PR TITLE
Fix compliance popover breaking other views presentation

### LINKED ISSUE #21758: Compliance Popover breaks presentation of views after onboarding
Compliance Popover breaks the presentation of views after onboarding. 

After tapping "Checking Stats" in onboard questions, the compliance popover either blocks the presentation of Stats or appears in a blank state.

### Expected behavior

Compliance Popover presentation should not block views appearing after onboarding and when the popover appears it shouldn't be in a blank state.

#### Atempt 1: Compliance Popover disappears after briefly showing
https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/c0a11338-f403-4330-9af6-34235aac5ac2

#### Atempt 2: Blank Compliance Popover opens up after opening Stats
https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/8abdede3-e34d-4e6e-ad47-b5201bec257d

### Steps to reproduce the behavior

1. Make sure you're in compliant country or  `shouldShowPrivacyBanner` returns `true`
2. Fresh install WordPress/Jetpack app
3. Log in
4. Click "Checking Stats" on onboarding questions
5. Notice that either Stats doesn't open up or if it opens, compliance popover appears blank

### PR DESCRIPTION
**Pull request by salimbraksa**
Fixes #21758

## Description
This PR fixes an issue where the Compliance Popover presentation could break other screens presentation.

## Test Instructions

1. Make sure you're in compliant country or  `shouldShowPrivacyBanner` returns `true`
2. Fresh install WordPress/Jetpack app
3. Log in
4. Click "Checking Stats" on onboarding questions
5. **Verify** the compliance popover appears without breaking Stats navigation

## Regression Notes

1. Potential unintended areas of impact
Compliance popover presentation.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift [hunk 1]
```diff
@@ -1,52 +1,86 @@
...
-    private weak var presentingViewController: UIViewController?
+
+    // MARK: - Dependencies
+
     private let complianceService = ComplianceLocationService()
     private let defaults: UserDefaults
 
+    // MARK: - Views
+
+    private static var window: UIWindow?
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift:18]: Initially, this variable wasn't `static`, but I changed that for the following reason:

There is another bug where 2 `MySiteViewController` are created. You can verify this by following the **Test Instructions** steps and using the **Memory Graph** debugger.

So when 2 `MySiteViewController` instances are created, 2 windows are also created and presented on screen. So to ensure this doesn't happened, I turned this variable `static`.

**Comment by mokagio** [comment]
I'm going to bump this to the next release because we'll be code freezing 23.8 today and this hasn't been approved yet.

File: WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift (hunk 1 above)
Comment (alpavanoglu) [reply WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift:18]: I suppose it should be fine as we wouldn't show this multiple times anyhow. Although, would it cause a problem if we were to reuse the View for some other purpose?

**Comment by alpavanoglu** [comment]
✅ Test Succeeds

**Review by alpavanoglu — APPROVED** [review]

File: WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift [hunk 2]
```diff
@@ -1,52 +1,86 @@
...
+
     private func shouldShowPrivacyBanner(countryCode: String) -> Bool {
         let isCountryInEU = Self.gdprCountryCodes.contains(countryCode)
         return isCountryInEU && !defaults.didShowCompliancePopup
     }
 
-    private func presentPopover(on viewController: UIViewController) {
+    private func dismiss(completion: (() -> Void)? = nil) {
+        self.presentingViewController.dismiss(animated: true) {
+            self.removeWindow()
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift:61]: `completion?()` call is missing here, making "Go to Settings" button non-functional.

**Review by staskus — APPROVED** [review]
Excuse me that the review took so long. I bookmarked it but somehow missed it.

⚠️  There's one change that's necessary to avoid regression. It's straightforward, so I added it myself to save some time since I'm already on a branch. 

Overall, the solution looks good and avoids many of the issues we would have if we try integrating this pop up into an existing navigation stack.

File: WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift (hunk 2 above)
Comment (staskus) [reply WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift:61]: Added it myself

**Pull request merged**